CodeAssignment_MVVM_Combine/Sources/Controllers/DetailViewController.swift

```diff
@@ -0,0 +1,309 @@
+    private lazy var backgroundCardView: UIView = {
+        let view = UIView()
+        view.clipsToBounds = true
+        view.layer.cornerRadius = 15
+        view.backgroundColor = .white
+        return view
+    }()
+    
+    /// リポジトリ名を表示するlabel
+    private lazy var repositoryNameLabel: UILabel = {
+        let label = UILabel()
+        label.font = .systemFont(ofSize: 20, weight: .semibold)
+        label.numberOfLines = 0
+        return label
+    }()
+    
+    /// Descriptionを表示するlabel
+    private lazy var descriptionLabel: UILabel = {
+        let label = UILabel()
+        label.font = .systemFont(ofSize: 18, weight: .regular)
+        label.numberOfLines = 0
+        return label
+    }()
+    
+    /// UserImageViewを入れるためのContentView
+    private lazy var userImageContentView: UIView = {
+        let view = UIView()
+        return view
+    }()
+    
+    /// Userのプロフィール画像
+    private lazy var userImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = .scaleAspectFill
+        imageView.image = UIImage(systemName: "person.circle")?.withTintColor(.systemMint, renderingMode: .alwaysOriginal)
+        imageView.clipsToBounds = true
+        imageView.layer.cornerRadius = 8
+        return imageView
+    }()
+    
+    /// User名を表示するためのlabel
+    private lazy var userNameLabel: UILabel = {
+        let label = UILabel()
+        label.textAlignment = .center
+        label.font = .systemFont(ofSize: 22, weight: .bold)
+        label.numberOfLines = 0
+        return label
+    }()
+    
+    /// 当該リポジトリのStarの数も一緒に表示するStarボタン
+    private lazy var starButton: UIButton = {
+        var config = UIButton.Configuration.filled()
+        let image = UIImage(systemName: "star")?.withTintColor(.systemGray3, renderingMode: .alwaysOriginal)
+        if let image {
+            config.image = image.withConfiguration(UIImage.SymbolConfiguration(pointSize: 25, weight: .regular))
+        }
+        config.baseBackgroundColor = .white
+        config.imagePlacement = .leading
+        // imageとtext間のSpace
+        config.imagePadding = 8
+        config.contentInsets = .init(top: 10, leading: 8, bottom: 10, trailing: 8)
+        config.cornerStyle = .large
+        
+        // borderLayerの設定
+        var backgroundConfig = UIBackgroundConfiguration.clear()
+        backgroundConfig.strokeColor = .systemGray2
+        backgroundConfig.strokeWidth = 2
+        config.background = backgroundConfig
+        
+        let button = UIButton(configuration: config)
+        // content hugging priorityを設定し、buttonのconstraintsが意図通りに設定されないことを防ぐ
+        button.setContentHuggingPriority(.required, for: .horizontal)
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        button.titleLabel?.setContentHuggingPriority(.required, for: .horizontal)
+        button.titleLabel?.setContentCompressionResistancePriority(.required, for: .horizontal)
+        
+        return button
```

Comment:
huggingPriorityとかはsetConstraintsのところでやったほうがいいかも。これも制約なので、場所が分かれてるとわかりずらくなっちゃいそう。

```suggestion        
        return UIButton(configuration: config)
```

Reply:
[a459314](https://github.com/KyusokLee/CodeAssignment_MVVM_Combine/pull/5/commits/a459314d2896307f45a692434d14e7b886a8fef6)にてコード修正しました！ご確認お願いします！